File: schemas/booking-scheduling/drizzle/service_categories.ts
// service_categories: hierarchical grouping of bookable services.
// See README.md for full design rationale.

import { pgTable, uuid, text, integer, boolean, timestamp, index } from "drizzle-orm/pg-core";

export const serviceCategories = pgTable(
  "service_categories",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    slug: text("slug").unique().notNull(),
    description: text("description"),
    parentId: uuid("parent_id").references(() => serviceCategories.id, { onDelete: "set null" }),
    position: integer("position").notNull().default(0),
    isActive: boolean("is_active").notNull().default(true),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_service_categories_parent_id").on(table.parentId),
    index("idx_service_categories_is_active_position").on(table.isActive, table.position),
  ]
);

File: schemas/booking-scheduling/spacetimedb/service_categories.rs
// service_categories: Hierarchical groupings for organizing bookable services.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = service_categories, public)]
pub struct ServiceCategory {
    #[primary_key]
    pub id: String, // UUID
    pub name: String,
    #[unique]
    pub slug: String,
    pub description: Option<String>,
    #[index(btree)]
    pub parent_id: Option<String>, // UUID — FK → service_categories.id (set null)
    pub position: i32,
    #[index(btree)]
    pub is_active: bool,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
    // Composite index: (is_active, position)
}
